src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+
+      Animal.alive = Animal.alive.filter(x => x.health > 0);
```

Comment:
reguarding to your question:
If you decide to use splice instead of filter take next into consideration and check for the indexOf result before doing splice: if indexOf returns (-1), that's to say the element wasn't found, splice will remove the last element of the array (negative indexing works).

Reply:
Thank you very much :+1: